jujuclient/setmodels_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jujuclient_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/jujuclient"
+	"github.com/juju/juju/testing"
+)
+
+type ControllerModelsSuite struct {
+	testing.FakeJujuXDGDataHomeSuite
+	store          jujuclient.ClientStore
+	controllerName string
+	controller     jujuclient.ControllerDetails
+}
+
+var _ = gc.Suite(&ControllerModelsSuite{})
+
+func (s *ControllerModelsSuite) SetUpTest(c *gc.C) {
+	s.FakeJujuXDGDataHomeSuite.SetUpTest(c)
+	s.store = jujuclient.NewFileClientStore()
+	s.controllerName = "test.controller"
+	s.controller = jujuclient.ControllerDetails{
+		ControllerUUID: "test.uuid",
+		APIEndpoints:   []string{"test.api.endpoint"},
+		CACert:         "test.ca.cert",
+		Cloud:          "aws",
+		CloudRegion:    "southeastasia",
+	}
+}
+
+func (s *ControllerModelsSuite) TearDownTest(c *gc.C) {
+	s.controller = jujuclient.ControllerDetails{}
+	s.controllerName = ""
+	s.store = nil
+	s.FakeJujuXDGDataHomeSuite.TearDownTest(c)
+}
+
+func (s *ControllerModelsSuite) TestSetModelsNoController(c *gc.C) {
+	err := s.store.SetModels(s.controllerName, nil)
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *ControllerModelsSuite) TestSetModelsNoControllerModels(c *gc.C) {
+	s.assertControllerStored(c)
```

Comment:
seeing as all but one of the tests want a controller, how about inverting it so you add a controller in SetUpTest, and then remove it in TestSetModelsNoController?

also, please rename "assertControllerStored" so that it says what it does. it *adds* a controller. if you just move the logic to the SetUpTest, you can remove the method altogether